### ISSUE TITLE
Outlier filtering

### ISSUE THREAD

**Issue by linusg** [owntracks/frontend#34]
I did, in fact, not take a trip into the River Thames.

![image](https://user-images.githubusercontent.com/19366641/81500642-70667280-92cb-11ea-8626-e651276e9693.png)

We could do this by calculating the average speed between points for example, a sudden spike is an outlier. (great idea from @growse)

Not sure if this should be part of the recorder's API or just a frontend thing.

**Comment by 0xspade** [comment]
This also happens to me, but we all know that this is a false positive, this also occurs in Traccar.

**Comment by rrkarman** [comment]
Would love to come up with a way to filter these outliers or possibly manually delete these from the dataset through the frontend interface. (I recently had one with a longitude of 0 (prime meridian) and an accuracy of 9 and I am in the US)

**Comment by linusg** [comment]
The recorder provides no API to delete locations, you'll have to do that manually or run a script on the `.rec` files.

**Comment by bryankaplan** [comment]
<img width="1370" height="779" alt="Image" src="https://github.com/user-attachments/assets/3131a05b-7c91-4979-adac-b169c86b70e7" />

Regrettably, I have not invented a teleporter.

I’ll write an outlier pruner, out of necessity. Am I right to suppose the server must be shut down before it modifies rec files? Are there any gotchas of which I should be aware?

**Comment by jpmens** [comment]
@bryankaplan the Recorder need not necessarily be shutdown if you are editing `.rec` files it doesn't have open, i.e. ones which it might write to at any moment. In case of doubt make a backup copy of the files you're about to edit so you can replace them if required.

Stopping the Recorder would be the safest method.

**Comment by bryankaplan** [comment]
After sleeping on this, I realize pruning must be ongoing for data received; it must therefore happen within recorder, while it’s running.

Yet for data received in the past, there needs to be an update mechanism.

Perhaps we should modify (not delete) records, with a flag like `outlier=true` so that a threshold can be later adjusted without data-loss.

**Comment by bryankaplan** [comment]
I believe this issue was actually resolved five years ago by b76cbdc2e627f7957a304bca5b454c20729181d6, which ostensibly closed only #35, but I posit should have closed this issue as well.

I updated my config using the suggested value from the docs: 

```
    filters: {
        minAccuracy: 100,
    },  
```

And presto, mine outliers hath gone poof.

**Comment by linusg** [comment]
No, accuracy filtering and outlier filtering are different, hence why I made two issues for them. You can have outliers with high accuracy like in the screenshot I posted. Accuracy filtering can hide a lot of garbage data but I found it's hard to find a value that works in all cases without removing too many legitimate datapoints too.

**Comment by bryankaplan** [comment]
I see! That's an important distinction.

From a UX standpoint, it would be nice to be able to click on any location and find a "delete" or "hide" link. Outliers should usually be infrequent enough that they could be manually pruned.

Do we know what causes a high-accuracy outlier, other than teleportation?

**Comment by linusg** [comment]
> From a UX standpoint, it would be nice to be able to click on any location and find a "delete" or "hide" link.

That would require an API in the recorder which currently doesn't exist. IIRC the stance on API-based history editing has been that it's unnecessarily complex and won't be supported.

> Do we know what causes a high-accuracy outlier, other than teleportation?

Most commonly wifi-based positioning, I've seen instances of being located across the country at events just because Google previously saw an access point elsewhere and associates it with its previous location. I also suspect that a temporary loss of signal can lead to bad data points that are not immediately assigned a low accuracy by the OS.

**Comment by bryankaplan** [comment]
Please forgive my naïvety — I'm new here — but what's so complex about adding an endpoint that accepts { user, device, timestamp } and which writes a prune file with the same name as the associated rec file, indicating a list of pruned records? That way there's no concern about modifying the presently open rec file.
